### PR TITLE
Exams

### PR DESCRIPTION
**Pull request by BbIXOD**

### CODE REVIEW

File: Tasks/replace.js [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
-      if (_index === -1) {
-        return res + src;
-      } else {
-        const start = src.substring(0, _index);
-        src = src.substring(_index + substr.length, src.length);
-        res += start + newstr;
+const replace = (str, substr, newStr) => {
+  if (substr === '') return str;
+  else {
+    let src = str;
```
Comment (erotourtes) [inline Tasks/replace.js:7]: Think about how you can rename src. It has a pretty cryptic name

File: Tasks/count.js [hunk 2]
```diff
@@ -1,16 +1,13 @@
...
-  }
-  keys = Object.keys(obj);
-  keys.forEach((key) => {
-    [];
-    const value = obj[key];
+const count = obj => {
+  let sum = 0;
+  const values = Object.values(obj);
+
+  for (const value of values)
```
Comment (erotourtes) [inline Tasks/count.js:8]: Think about `arr.reduce()` instead of `for loop`

**Review by erotourtes — CHANGES_REQUESTED** [review]
It looks like you need to change the naming in `replace.js`. Also, think about `reduce`

File: Tasks/count.js (hunk 2 above)
Comment (BbIXOD) [reply Tasks/count.js:8]: This is a good idea. Maybe I will do another branch with this realization later. But this changes are too radical

File: Tasks/replace.js (hunk 1 above)
Comment (BbIXOD) [reply Tasks/replace.js:7]: Is `source` a good one?

**Comment by BbIXOD** [comment]
> It looks like you need to change the naming in `replace.js`. Also, think about `reduce`

File: Tasks/replace.js (hunk 1 above)
Comment (erotourtes) [reply Tasks/replace.js]: Maybe add `sourceStr` for clarity

**Review by erotourtes — CHANGES_REQUESTED** [review]

**Review by erotourtes — APPROVED** [review]

File: Tasks/replace.js (hunk 1 above)
Comment (erotourtes) [reply Tasks/replace.js]: I'd prefer `sourceStr`

File: Tasks/replace.js [hunk 3]
```diff
@@ -1,22 +1,21 @@
...
-      const _index = src.indexOf(substr);
-      if (_index === -1) {
-        return res + src;
-      } else {
-        const start = src.substring(0, _index);
-        src = src.substring(_index + substr.length, src.length);
-        res += start + newstr;
+const replace = (str, substr, newStr) => {
+  if (substr === '') return str;
+  else {
```
Comment (erotourtes) [inline Tasks/replace.js:6]: You don't need `else` block

**Review by erotourtes — CHANGES_REQUESTED** [review]

**Comment by BbIXOD** [comment]
Filter + reduce are slower and more complicated then for .. of.

File: Tasks/replace.js [hunk 4]
```diff
@@ -2,20 +2,24 @@
...
-    let res = '';
-    while (true) {
-      const index = source.indexOf(substr);
-      if (index === -1) return res + source;
-      else {
-        const start = source.substring(0, index);
-        source = source.substring(index + substr.length, source.length);
-        res += start + newStr;
-      }
+  if (substr === '') {
```
Comment (erotourtes) [inline Tasks/replace.js:5]: The parentheses are redundant

File: Tasks/replace.js [hunk 5]
```diff
@@ -2,20 +2,24 @@
...
+    return str;
+  }
+
+  let source = str;
+  let res = '';
+
+  while (true) {
```
Comment (erotourtes) [inline Tasks/replace.js:12]: I'd recommend you assign `res` with the input string, and get rid of `source`. Also in js, there is replace method `str.replace(subStr, newStr)`, you can use that, and then in your `while` loop, you won't need to manipulate with strings by hand. Also, consider `str.includes(subStr)` to check in while loop is there is any substrings to replace

**Review by erotourtes — CHANGES_REQUESTED** [review]
I think you can further improve your code

File: Tasks/replace.js (hunk 5 above)
Comment (erotourtes) [reply Tasks/replace.js:12]: As it was found out, you need your custom implementation. ILGTM, good luck!

**Pull request merged**